sql/src/main/java/io/crate/metadata/Functions.java

```diff
@@ -148,35 +149,78 @@ public FunctionImplementation getUserDefined(String schema,
      * Returns the user-defined function implementation for the given function name and arguments.
      * The types may be cast to match the built-in argument types.
      *
-     * @param name The function name.
+     * @param functionQualifiedName  The function qualified name. This can be in the form of `schema.functionName` or
+     *                               `functionName`.
      * @param arguments The function arguments.
+     * @param searchPath The {@link SearchPath} against which to try to resolve the function if it is not identified by
+     *                   a fully qualifed name (ie. `schema.functionName`)
      * @return a function implementation.
      */
     @Nullable
-    public FunctionImplementation getUserDefinedByArgs(String schema,
-                                                       String name,
-                                                       List<? extends FuncArg> arguments) throws UnsupportedOperationException {
-        FunctionResolver resolver = lookupUdfFunctionResolver(schema, name, arguments);
+    public FunctionImplementation resolveUserDefinedByArgs(QualifiedName functionQualifiedName,
```

Comment:
couldn't we keep schema and name as arguments here? it looks like we're creating QualifiedName right at the call site just to unpack it immediately

Reply:
Yes we can, was looking to keep the `resolve...` methods consistent